src/Components/Server/test/Circuits/CircuitIdFactoryTest.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components.Server.Tests.Circuits
+{
+    public class CircuitIdFactoryTest
+    {
+        [Fact]
+        public void CreateCircuitId_Generates_NewRandomId()
+        {
+            CircuitIdFactory factory = CreateTestFactory();
+
+            // Act
+            var id = factory.CreateCircuitId();
+
+            // Assert
+            Assert.NotNull(id);
+            // This is the magic data protection header that validates its protected
+            Assert.StartsWith("CfDJ", id);
```

Comment:
Do we really want this hardcoded here? This seems bad. Do we have other options?

Reply:
Its quite the opposite. That is a good test that things are being data protected. You'll see that prefix on every data protected payload.